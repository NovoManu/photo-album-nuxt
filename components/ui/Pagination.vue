<template>
  <v-footer fixed height="56px" dark class="my-pagination">
    <v-layout row wrap justify-end align-center>
      <div class="d-flex align-center">
        <div class="hidden-xs-only">Rows per page:</div>
        <v-card flat tile max-width="50px" color="transparent" class="mx-3">
          <v-select
            :value="limit"
            :items="limits"
            hide-details
            single-line
            class="pt-0"
            @change="$emit('changed', $event)"
          />
        </v-card>
        <v-btn icon :disabled="firstPage" @click="$emit('decrement')">
          <v-icon>chevron_left</v-icon>
        </v-btn>
        <v-btn icon :disabled="lastPage" @click="$emit('increment')">
          <v-icon>chevron_right</v-icon>
        </v-btn>
      </div>
    </v-layout>
  </v-footer>
</template>
<script lang="ts">
import {
  Component,
  Prop,
  Vue,
  Model
} from '~/node_modules/nuxt-property-decorator'

@Component
export default class Pagination extends Vue {
  @Prop({ required: true }) firstPage!: boolean
  @Prop({ required: true }) lastPage!: boolean
  @Model('changed') limit!: number
  limits: number[] = [20, 30, 50]
}
</script>
