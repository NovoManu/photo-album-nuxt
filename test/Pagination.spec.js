import { mount } from '@vue/test-utils'
import Pagination from '@/components/ui/Pagination.vue'

describe('Pagination.vue', () => {
  it('renders component', () => {
    const wrapper = mount(Pagination)
    expect(wrapper.element).toMatchSnapshot()
  })
})
