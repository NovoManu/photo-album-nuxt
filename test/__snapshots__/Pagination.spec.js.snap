// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination.vue renders component 1`] = `
<v-footer
  class="my-pagination"
  dark=""
  fixed=""
  height="56px"
>
  <v-layout
    align-center=""
    justify-end=""
    row=""
    wrap=""
  >
    <div
      class="d-flex align-center"
    >
      <div
        class="hidden-xs-only"
      >
        Rows per page:
      </div>
       
      <v-card
        class="mx-3"
        color="transparent"
        flat=""
        max-width="50px"
        tile=""
      >
        <v-select
          class="pt-0"
          hide-details=""
          items="20,30,50"
          single-line=""
        />
      </v-card>
       
      <v-btn
        icon=""
      >
        <v-icon>
          chevron_left
        </v-icon>
      </v-btn>
       
      <v-btn
        icon=""
      >
        <v-icon>
          chevron_right
        </v-icon>
      </v-btn>
    </div>
  </v-layout>
</v-footer>
`;
